v1/nix/pkgs/fetchPipMetadata/src/fetch_pip_metadata/lock_file_from_report.py

```diff
@@ -10,13 +10,17 @@
 )
 
 
-def git_repo_root():
-    return subprocess.run(
+def git_repo_root(directory):
+    proc = subprocess.run(
         ["git", "rev-parse", "--show-toplevel"],
-        check=True,
         text=True,
         stdout=subprocess.PIPE,
-    ).stdout.strip()
+        cwd=directory,
+    )
+    # fall back to the current directory if not a git repo
+    if proc.returncode == 128:
+        return str(Path(".").absolute())
```

Comment:
This is required to make it work outside of git repos, like for example in the nix sandbox during `checkPhase`.

Reply:
Commented here by mistake https://github.com/nix-community/dream2nix/pull/569#discussion_r1258094079